Library/Homebrew/cmd/install.rb

```diff
@@ -40,20 +40,28 @@ def install
       end
 
       ARGV.formulae.each do |f|
-        # Building head-only without --HEAD is an error
-        if not ARGV.build_head? and f.stable.nil?
+        # head-only without --HEAD is an error
+        if not ARGV.build_head? and f.stable.nil? and f.devel.nil?
           raise <<-EOS.undent
           #{f.name} is a head-only formula
           Install with `brew install --HEAD #{f.name}`
           EOS
         end
 
-        # Building stable-only with --HEAD is an error
+        # devel-only without --devel is an error
+        if not ARGV.build_devel? and f.stable.nil? and f.head.nil?
```

Comment:
Requiring `f.head.nil?` means that a formula with devel and head but no stable will not generate the proper error, so this should probably be more like

```rb
if not ARGV.build_devel? and f.stable.nil?
  if f.head.nil?
    raise <<-EOS.undent
      #{f.name} is a devel-only formula
      Install with `brew install --devel #{f.name}`
      EOS
  else
    raise "#{f.name} has no stable download, please choose --devel or --HEAD"
  end
end
```